### ISSUE TITLE
cuda

### ISSUE THREAD

**Issue by yuanhs1996** [AllentDan/SegmentationCpp#32]
Can training be accelerated using CUDA? If so, how can it be done?

**Comment by AllentDan** [comment]
Yes, examples are in the readme.
```cpp
Segmentor<FPN> segmentor;
segmentor.Initialize(0/*gpu id, -1 for cpu*/,
                    512/*resize width*/,
                    512/*resize height*/,
                    {"background","person"}/*class name dict, background included*/,
                    "resnet34"/*backbone name*/,
                    "your path to resnet34.pt");
segmentor.Train(0.0003/*initial leaning rate*/,
                300/*training epochs*/,
                4/*batch size*/,
                "your path to voc_person_seg",
                ".jpg"/*image type*/,
                "your path to save segmentor.pt");
```

**Comment by yuanhs1996** [comment]
thank you



---Original---
From: ***@***.***&gt;
Date: Thu, Mar 7, 2024 10:14 AM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [AllentDan/LibtorchSegmentation] cuda (Issue #32)




 
Closed #32 as completed.
